src/python/grpcio/grpc/_common.py

```diff
@@ -76,9 +76,41 @@
 }
 
 
-def metadata(application_metadata):
+def encode(s):
+  if isinstance(s, bytes):
+    return s
+  else:
+    return s.encode('ascii')
```

Comment:
Definitely not a stupid question:
In all the places where we encode on behalf of the user, the string is required to an ascii string, ie method names/message details/metadata keys/values.  This will fail fast if they don't meet they don't satisfy that contract.